src/caret_analyze/common/type_check_decorator.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import wraps
+from inspect import Signature, signature
+from typing import Any, Dict, List, Tuple
+
+from ..exceptions import UnsupportedTypeError
+
+
+try:
+    from pydantic import validate_arguments, ValidationError
+
+    def _get_expected_types(e: ValidationError) -> str:
+        """
+        Get expected types.
+
+        Parameters
+        ----------
+        e: ValidationError
+            ValidationError instance has one or more ErrorDict instances.
+            Example of ErrorDict structure is as follows.
+            (i) Build-in type case:
+                {'type': 'type_error.[EXPECT_TYPE]', ...}
+
+            (ii) Custom class type case:
+                {'type': 'type_error.arbitrary_type',
+                'ctx': {'expected_arbitrary_type': '[EXPECT_TYPE]'}, ...}
+
```

Comment:
Could you add descriptions for return values?

```suggestion

        Returns
        -------
        str
            _description_
            
```

I'm using this extension: :smiley: 
https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring